crates/wysiwyg/src/composer_model/selection.rs

```diff
@@ -62,6 +63,34 @@ where
             (s, e)
         }
     }
+
+    /// Return a boolean to let us know if we have a selection
+    pub fn has_selection(&self) -> bool {
+        let (s, e) = self.safe_selection();
+        s != e
+    }
+
+    /// Return a boolean to let us know if we have a cursor, ie a zero length selection
+    pub fn has_cursor(&self) -> bool {
+        let (s, e) = self.safe_selection();
+        s == e
+    }
+
+    /// Return a boolean to let us know if we are touching the edge of the DOM
+    pub fn has_selected_to_end_of_dom(&self) -> bool {
+        let (s, e) = self.safe_selection();
+        s == 0 || e == self.state.dom.text_len()
+    }
+
+    /// Return a boolean to let us know if we are touching the edge of the DOM
+    /// when moving in a specific direction
+    pub fn has_selected_end_of_dom(&self, direction: &Direction) -> bool {
```

Comment:
Surprising that we need this as well as has_selected_to_end_of_dom but if you need it, that's fine.

Reply:
We don't! Think I missed that when trying to pick the changes over